spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -580,7 +598,7 @@ public Void doInRabbit(Channel channel) throws Exception {
 						return null;
 					}
 				});
-				this.usingFastReplyTo = true;
+				return true;
 			}
 			catch (Exception e) {
 				if (this.replyAddress != null) {
```

Comment:
No, it's also called if `|| Address.AMQ_RABBITMQ_REPLY_TO.equals(this.replyAddress)`, per line 591.

i.e. if the user explicitly attempted to configure direct reply-to while using a broker that doesn't support it.